### ISSUE TITLE
Ycheck failures after frontend

### ISSUE THREAD

**Issue by DarkDimius** [lampepfl/dotty#725]
Some tests fail Ycheck after frontend

```
./tests/pos/java-interop/t2413/TestScalac.scala
./tests/pos/jon.scala
./tests/run/Course-2002-01.scala
```

**Comment by smarter** [comment]
All of the tests mentioned now pass Ycheck:frontend, but we should check if others don't and if we can reenable Ycheck:all.

**Comment by allanrenucci** [comment]
I recently tried to `Ycheck:all` and we still have failure after frontend for:
- pos/t1513a.scala
- pos/t1513b.scala
- pos/reference/named-typeargs.scala
- tests/run/t7126.scala
- tests/run/hmap.scala
- tests/run/hmap-covariant.scala
- compiler/src/dotty/tools/dotc/core/Hashable.scala
- compiler/src/dotty/tools/dotc/transform/FunctionalInterfaces.scala
- collection-strawman/collections/src/main/scala/strawman/collection/BitSet

**Comment by allanrenucci** [comment]
Closed in #4183
